orangewidget/gui.py

```diff
@@ -1083,8 +1083,42 @@ def appendRadioButton(group, label, insertInto=None,
     return w
 
 
+class DelayedNotification(QObject):
+    """
+    A proxy for successive calls/signals that emits a signal
+    only when there are no calls for a given time.
+
+    Also allows for mechanism that prevents successive equivalent calls:
+    ff values are passed to the "changed" method, a signal is only emitted
+    if the last passed values differ from the last passed values at the
+    previous emission.
+    """
+    notification = Signal()
+
+    def __init__(self, parent=None, timeout=500):
+        super().__init__(parent=parent)
+        self.timeout = timeout
+        self._timer = QTimer(self)
+        self._timer.timeout.connect(self.notify_immediately)
+        self._did_notify = False  # if anything was sent at all
+        self._last_value = None  # last value passed to changed
+        self._last_notified = None  # value at the last notification
+
+    def changed(self, *args):
+        self._last_value = args
+        self._timer.start(self.timeout)
+
+    def notify_immediately(self):
+        self._timer.stop()
+        if self._did_notify and self._last_notified == self._last_value:
+            return
+        self._last_notified = self._last_value
+        self._did_notify = True
+        self.notification.emit()
+
+
 def hSlider(widget, master, value, box=None, minValue=0, maxValue=10, step=1,
-            callback=None, label=None, labelFormat=" %d", ticks=False,
+            callback=None, callback_finished=None, label=None, labelFormat=" %d", ticks=False,
```

Comment:
Do you think there will be situations when a user will want to specify different functions for `callback` _and_ `callback_finished`?
Because otherwise it might be better to just keep using `callback` and introduce a new parameter like `delay` or something similar, where you can specify e.g. 500 (ms). If it is None you get the old functionality, and if you provide a delay the callback is used like callback_finished in your new version.
(The `delay` parameter could also be just `True`/`False` in you don't think being able to choose a different length is useful)

Reply:
Some sliders show chosen values in label, like the slider in the Logistic Regression widget. Especially there, where it is non-linear, it is good to see what we are choosing. I could not think of any other way that to have two callbacks.

Reply:
I agree with @markotoplak, sliders often update some label instantly, but should update the widget when the action is finished.

But can't we achieve this by connecting `sliderMoved` to `callback` and slider released to `callback_finished`? Do we need this delay?

Reply:
@janezd, if we used sliderReleased as you suggested we would have to handle keyboard presses (at least on my system I can focus the slider and control it with a keyboard) and mouse wheel events. The current implementation calls callback_finished on sliderReleased without the delay.

Reply:
Yes, in that case two callbacks seem fine.
The delay itself would probably not be worth it as it is useful only in rare specific circumstances. But if it helps to simplify the implementation, I guess we get a bonus feature alongside not having to worry about many different event types.